src/dotnet/Program.cs

```diff
@@ -140,6 +140,8 @@ private static int ProcessArgs(string[] args)
             {
                 // mimic the env variable
                 Environment.SetEnvironmentVariable(CommandContext.Variables.Verbose, verbose.ToString());
+                // Use the current value of AnsiPassThru but make sure child processes have it set to true.
+                CommandContext.ShouldPassAnsiCodesThrough();
```

Comment:
The intent to cause CommandContext to read the current value of the environmental variable before changing it on the following line. The intention is for the current process to use the value of the DOTNET_CLI_CONTEXT_ANSI_PASS_THRU as it is when the process starts but for all the children to have it set to true. Otherwise dotnet.exe will print out a bunch of escape codes on Windows because AnsiPassThrough will always be true.

Since calling that getter method for it's side effect is obtuse, perhaps I should add a new function to CommandContext that reads the environmental variable. Something like "InitialAnsiPassThroughFromEnvironment()". What do you think?